contrib/postprocessing/lethe_pyvista_tools.py

```diff
@@ -100,87 +128,121 @@ def __init__(self, case_path = ".", prm_file_name = ""):
             self.first = first
             self.step = step
             self.last = len(self.time_list) - 1
         else:
-            self.list_vtu = self.list_vtu[first:last:step]
+            list_vtu = list_vtu[first:last:step]
             self.time_list = self.time_list[first:last:step]
             self.first = first
             self.step = step
             self.last = last
 
-        # Create empty array to store results
-        self.df = []
+        # List of paths among read data
+        read_files_path_list = [pvd_datasets[x].path for x in range(len(pvd_datasets))]
 
-        # Read VTU data
-        N_vtu = len(self.list_vtu)
-        pbar = tqdm(total = N_vtu, desc="Reading VTU files")
-        for i in range(len(self.list_vtu)):
-            
-            # Read dataframes from VTU files into df
-            self.df.append(pv.read(f"{self.path_output}/{self.list_vtu[i]}"))
+        # Write new vtu and pvd files to store modified data.
+        # IMPORTANT!!!! If this parameter is empty, that is,  "", data will be written over the original vtu and pvd files.
+        with open(f'{self.path_output}/{pvd_name}') as pvd_in:
+            with open(f'{self.path_output}/{prefix}{pvd_name}', 'w') as pvd_out:
+                for line in pvd_in:
+                    
+                    # If line refers to a dataset
+                    if "vtu" in line:
+
+                        # For all read files
+                        for path in read_files_path_list:
+
+                            # If line matches one of the files
+                            if path in line:
+                                line = line.replace('.pvtu', '.0000.vtu')
+                                line = line.replace('file="', f'file="{prefix}')
+                                pvd_out.write(line)
+                                read_files_path_list.remove(path)
+                                pass
+                    
+                    # Write config lines
+                    else:
+                        pvd_out.write(line)
+
+        # Make a copy of VTU files
+        N_vtu = len(list_vtu)
+        pbar = tqdm(total = N_vtu, desc="Writting modified VTU and PVD files")
+        self.list_vtu = []
+        for i in range(len(list_vtu)):
+            # Copy file
+            shutil.copy2(f'{self.path_output}/{list_vtu[i]}', f'{self.path_output}/{prefix}{list_vtu[i]}')
+
+            # Append to list of names of VTU files
+            self.list_vtu.append(f'{prefix}{list_vtu[i]}')
             pbar.update(1)
 
-        print(f'Written .df[timestep] from timestep = 0 to timestep = {len(self.list_vtu)-1}')
+        # Fix name of PVD file
+        self.pvd_name = prefix + pvd_name
+
+        # Create pyvista reader for files in the new .pvd file 
+        self.reader = pv.get_reader(f"{self.path_output}/{self.pvd_name}") 
+
+        # Create list of PVD datasets with new files
+        self.pvd_datasets = self.reader.datasets
```

Comment:
This is kinda the same as lines 137 - 140, is it normal?

Reply:
It is because I update it with the new data.